systemtest/src/main/java/io/strimzi/systemtest/Resources.java

```diff
@@ -511,7 +511,7 @@ private Kafka waitFor(Kafka kafka) {
         StUtils.waitForAllStatefulSetPodsReady(KafkaResources.kafkaStatefulSetName(name), kafka.getSpec().getKafka().getReplicas());
         LOGGER.info("Kafka pods are ready");
         // EO should not be deployed if it does not contain UO and TO
-        if (kafka.getSpec().getEntityOperator() != null) {
+        if (kafka.getSpec().getEntityOperator() != null && kafka.getSpec().getEntityOperator().equals(new Object())) {
```

Comment:
What is this supposed to do? I ask because AFAICS `kafka.getSpec().getEntityOperator().equals(new Object())` will always return false.

Reply:
Good point, it should be fixed now to check if UO and TO are null.